app/views/list_proposal.html

```diff
@@ -64,32 +64,29 @@
           <a href="#/new-proposal" class="col-md-offset-1">
             <button type="button" class="btn btn-success "><i class="fa fa-plus"></i> Додати пропозицію</button>
           </a>
-          <button type="button" class="pull-right btn btn-primary"><i class="fa fa-bars "></i></button>
+          <button type="button" class="pull-right btn btn-primary" ng-click="open('sm')"><i class="fa fa-bars "></i></button>
         </div>
 
         <!-- Table -->
-          <table ng-table="tableParams" class="table">
+        <div>
+          <table ng-table="tableParams" show-filter="true" class="table">
+            <thead>
+            <tr>
+              <td ng-repeat="column in headers" data-title="'column.display'" ng-show="column.visible" sortable="column.field">
+                {{column.display}}
+              </td>
+            </tr>
+            </thead>
             <tr ng-repeat="prop in $data">
-              <td data-title="'specialityID'">{{prop.specialtyId}}</td>
-              <td data-title="'departmentID'">{{prop.departmentId}}</td>
-              <td data-title="'timePeriodId'">{{prop.timePeriodId}}</td>
-              <td data-title="'timePeriodCourseId'">{{prop.timePeriodCourseId}}</td>
-              <td data-title="'specofferTypeId'">{{prop.specofferTypeId}}</td>
-              <!--<td data-title="'docSeries'">{{prop.docSeries}}</td>-->
-              <!--<td data-title="'docNum'">{{prop.docNum}}</td>-->
-              <td data-title="'eduFormTypeId'">{{prop.eduFormTypeId}}</td>
-              <td data-title="'licCount'">{{prop.licCount}}</td>
-              <td data-title="'stateCount'">{{prop.stateCount}}</td>
-              <!--<td data-title="'begDate'">{{prop.begDate}}</td>-->
-              <!--<td data-title="'endDate'">{{prop.endDate}}</td>-->
-              <!--<td data-title="'parentId'">{{prop.parentId}}</td>-->
+              <td ng-repeat="column in headers" data-title="'column.display'" ng-show="column.visible" sortable="column.field">
```

Comment:
how "sortable" attribute works? Do we need it here and in header?
and again: your tast is not about sorting. but filtering columns